messenger/http_transporter.go

```diff
@@ -124,13 +139,12 @@ type HTTPTransporter struct {
 
 // NewHTTPTransporter creates a new http transporter with an optional binding address.
 func NewHTTPTransporter(upid upid.UPID, address net.IP) *HTTPTransporter {
-	tr := &http.Transport{}
 	result := &HTTPTransporter{
 		upid:         upid,
 		messageQueue: make(chan *Message, defaultQueueSize),
 		mux:          http.NewServeMux(),
-		client:       &http.Client{Transport: tr},
-		tr:           tr,
+		client:       httpClient,
+		tr:           httpTransport,
```

Comment:
You'd have to cast it because http.Client.Transport is a RoundTripper
interface. I'd rather not cast.

On Thu, Oct 15, 2015 at 8:45 PM, Karl Isenberg <notifications@github.com>
wrote:

> In messenger/http_transporter.go
> <https://github.com/mesos/mesos-go/pull/198#discussion_r42199089>:
>
> >  	result := &HTTPTransporter{
> >  		upid:         upid,
> >  		messageQueue: make(chan *Message, defaultQueueSize),
> >  		mux:          http.NewServeMux(),
> > -		client:       &http.Client{Transport: tr},
> > -		tr:           tr,
> > +		client:       httpClient,
> > +		tr:           httpTransport,
>
> Why is the transport supplied to the HTTPTransporter if you can just pull
> it out of the Client with t.client.Transport?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/198/files#r42199089>.
>